File: android/app/src/main/java/com/example/greetings/GreetingsActivity.java
package com.example.greetings;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class GreetingsActivity extends AppCompatActivity {

    static {
        System.loadLibrary("rust");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_greetings);

        RustGreetings g = new RustGreetings();
        String r = g.sayHello(", this comes from Rust");
        ((TextView)findViewById(R.id.greetingField)).setText(r);
    }
}

File: android/app/src/main/java/com/example/greetings/GreetingsActivity.kt
package com.example.greetings

import androidx.appcompat.app.AppCompatActivity

import android.os.Bundle
import android.view.View
import android.widget.TextView

class GreetingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_greetings)

        val g = RustGreetings()
        val r = g.sayHello(", this comes from Rust")
        val textView = findViewById<View>(R.id.greetingField) as TextView
        textView.text = r
    }

    companion object {

        init {
            System.loadLibrary("rust")
        }
    }
}
